### ISSUE TITLE
Consolidate meta (metadata) information

### ISSUE THREAD

**Issue by char0n** [swagger-api/apidom#867]
These are the agreed rules on engagement about adding meta (metadata) to ApiDOM:

- when the meta information is **static** and generic to all usecases and we  have a concrete element (CE) type: we add it to element constructors
- when the meta information is **static** and generic to all usecases and we have a non-concrete element (NCE)  type: we add it to element constructors
- in all other cases metadata gets added in Visitors in refractor layer

---

**Action items:**

- [x] identify and refactor places where these rules are broken
- [x] review `version` metadata

**Comment by char0n** [comment]
@frantuma pinging about the decision here.

**Comment by frantuma** [comment]
@char0n I am not sure here, possibly it would be good to sync in a chat; I understand your points, let's say that possibly keeping things as they are could be also an option, following your original direction (static vs refactor, and defined NCEs). In your example above, we might want to add metadata in refractors for `Array<Message>` which is kind of "outside the pattern" but at least for the moment I would say that this is acceptable and also kind of more correct in terms of static vs "use case dependent"

**Comment by char0n** [comment]
Right, so if I understand correctly, we're keeping the status quo until we decide to change it in the future.
